### PR TITLE
DCES-67: LocalApp Authentication Config and local CLI for Demo

### PR DESCRIPTION
**Pull request by tanovellino**
…cess too

## What

[Link to story](https://dsdmoj.atlassian.net/browse/LASB-XXX)

Describe what you did and why.

## Checklist

Before you ask people to review this PR:

- [x] Tests should be passing: `./gradlew test`
- [x] Github should not be reporting conflicts; you should have recently run `git rebase main`.
- [x] Avoid mixing whitespace changes with code changes in the same commit. These make diffs harder to read and conflicts more likely.
- [x] You should have looked at the diff against main and ensured that nothing unexpected is included in your changes.
- [x] You should have checked that the commit messages say why the change was made.

### CODE REVIEW

**Comment by LAA-Dave-Edwards** [comment]
Can you also look at the sonar issues, there's coverage and some basic smells.

**Comment by LAA-Dave-Edwards** [comment]
Can you also fill out the description for the PR?

File: dces-report-service/src/main/java/uk/gov/justice/laa/crime/dces/utils/DateUtils.java [hunk 1]
```diff
@@ -30,4 +30,12 @@ public boolean validateDate(XMLGregorianCalendar date, LocalDate startDate, Loca
                 && convertedDate.compareTo(endDate)<=0);
     }
 
+    public static LocalDate getCurrentFromDate() {
```
[the remark this answers is not in the document]
Comment (tanovellino) [reply dces-report-service/src/main/java/uk/gov/justice/laa/crime/dces/utils/DateUtils.java:33]: I'll change getDeafultStartDate, also do it for the toDate

File: dces-report-service/src/main/java/uk/gov/justice/laa/crime/dces/report/functions/DcesReportFunctionsFactory.java [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+
+            try {
+                File reportFile = reportService.getFdcReport(fromDate, toDate);
+            } catch (JAXBException | IOException e) {
+                throw new RuntimeException(e);
+            }
+        };
+    }
+
+    private boolean validateInputs(MessageHeaders headers) {
```
[the remark this answers is not in the document]
Comment (tanovellino) [reply dces-report-service/src/main/java/uk/gov/justice/laa/crime/dces/report/functions/DcesReportFunctionsFactory.java:69]: I changed it to `shouldUseDefaultDates`

File: dces-report-service/src/main/java/uk/gov/justice/laa/crime/dces/report/functions/DcesReportFunctionsFactory.java [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+
+    private LocalDate convertToDate(MessageHeaders headers) {
+        return (validateInputs(headers)) ?
+            convertDateFromHeader("todate", headers) :
+                DateUtils.getCurrentToDate()
+        ;
+    }
+
+    private LocalDate convertDateFromHeader(String key, MessageHeaders headers) {
+        if (!headers.containsKey(key))
```
[the remark this answers is not in the document]
Comment (tanovellino) [reply dces-report-service/src/main/java/uk/gov/justice/laa/crime/dces/report/functions/DcesReportFunctionsFactory.java:88]: The idea is that this method was more "generic" or not be tied to the previous two. Hence the sanity check I left there. But will remove in this case anyway the method is private and only called for this

File: dces-report-service/src/main/java/uk/gov/justice/laa/crime/dces/report/maatapi/MaatApiWebClientFactory.java [hunk 4]
```diff
@@ -69,8 +69,8 @@ public WebClient maatApiWebClient(
 //            clientBuilder.defaultHeader(
```
[the remark this answers is not in the document]
Comment (tanovellino) [reply dces-report-service/src/main/java/uk/gov/justice/laa/crime/dces/report/maatapi/MaatApiWebClientFactory.java:69]: added a TODO comment to resolve later

File: dces-report-service/src/main/java/uk/gov/justice/laa/crime/dces/utils/DateUtils.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (tanovellino) [reply dces-report-service/src/main/java/uk/gov/justice/laa/crime/dces/utils/DateUtils.java:33]: changed to `getDefaultStartDateForReport`

**Comment by tanovellino** [comment]
Will add test coverage in a following merge request

**Review by atharmajeed — APPROVED** [review]

**Pull request merged**